print_message_to_players() tweaks

TheBigZet opened the pull request:
### Brief description of the changes:
1) Tightened the loop by iterating only on connected players
2) Removed `colorChosen == nil` check, as `print(msg, {color = nil})` works just fine
### Tested Changes:
- [ ] I've tested the changes locally or with people.
- [ ] I've not tested the changes.

cogito123 commented:
When I leave and rejoin server afterwards I see messages that have appeared during my absence. If we merge this PR, is it going to remove this behavior?

TheBigZet commented:
> When I leave and rejoin server afterwards I see messages that have appeared during my absence. If we merge this PR, is it going to remove this behavior?

Git diff doesn't show this well enough, but originally `print_message_to_players()` checked if player is connected before printing to him
https://github.com/Factorio-Biter-Battles/Factorio-Biter-Battles/blob/9f486bab51c4e50718b1ce975c1edb640595fdd7/maps/biter_battles_v2/functions.lua#L478

I removed that check and replaced it with passing list of connected players directly to `forcePlayerList`
https://github.com/Factorio-Biter-Battles/Factorio-Biter-Battles/blob/9f486bab51c4e50718b1ce975c1edb640595fdd7/maps/biter_battles_v2/functions.lua#L475

https://github.com/Factorio-Biter-Battles/Factorio-Biter-Battles/blob/922d2d89432f3468ec4304db6a1616f09433af74/commands/chat_commands.lua#L33-L34

This PR is only related to chatting with teams via `/nth`, `/sth`, `/spectator-chat` etc.
I believe you're referencing normal messages, which I guess are done as `game.print()` by the engine itself by default
